### ISSUE TITLE
Address bar on the bottom is obscuring the bottom part of pages.

### ISSUE THREAD

**Issue by gcp** [mozilla-mobile/fenix#13797]
![Screenshot_20200810-215514](https://user-images.githubusercontent.com/1104290/90145162-ee7d5e00-dd7f-11ea-886c-26a492a56007.png)

## Steps to reproduce

1) Load intervals.icu (probably needs a Strava account to do anything useful)
2) Load an activity with heart rate data
3) Visit the "heart" page

### Expected behavior

Can read text on the page.

### Actual behavior

Text on the bottom of the page is obscured by the address bar. Swapping the address bar to the top reveals the text.

### Device information

* Android device: Pixel 2 XL
* Fenix version: Nightly

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-4974)

**Comment by liuche** [comment]
We should investigate this, since there are lots of issues being filed about these: #14377 #14386 #14361

Sites mentioned are:
https://discourse.mozilla.org/t/add-on-support-in-new-firefox-for-android/53488/25
https://www.homedepot.com
proton mail

**Comment by mattok** [comment]
Just a note that I too have observed the buttons/content being cut off at the bottom of the page, in contexts where `position: sticky` is used (#14421) and in contexts where I could not identify any use of `position: sticky`. For example, this happened to me for personal data consent message on https://www.thetimes.co.uk/

**Comment by Mugurell** [comment]
Thank you, we'll investigate.

**Comment by Mugurell** [comment]
I think most of the issues here should have been resolved in the latest Nightly by https://bugzilla.mozilla.org/show_bug.cgi?id=1633322
Now on all sites a swipe up when the page is at the bottom should hide the toolbar and show the bottom of the page.
I see the issues being resolved for the above `discourse` link and no issue for the consent dialog from `thetimes.co.uk`
Could not access `homedepot`, `offerup` to test.

![WebpageCutoff](https://user-images.githubusercontent.com/11428869/91851207-752ea800-ec67-11ea-8ba9-d4f86c74be16.gif)
[video](https://drive.google.com/file/d/1ClImvxghvZOiuJU2riaWefVuxc6Vk4vC/view?usp=sharing)

@ QA Have you seen this still reproducing in latest nightly?

If the issue is still encountered please post the exact page url and we'll investigate.

**Comment by yoasif** [comment]
@Mugurell FYI Offerup is still not working correctly.

**Comment by ffxsam** [comment]
@Mugurell The scrolling does indeed hide it, but in some cases it's not possible to scroll. E.g. if I'm on a responsive web app on a page that doesn't have any content that extends outside the view. Then I'm stuck, potentially with UI elements at the bottom of the page being blocked by the address bar.

**Comment by theres-waldo** [comment]
> @Mugurell The scrolling does indeed hide it, but in some cases it's not possible to scroll. E.g. if I'm on a responsive web app on a page that doesn't have any content that extends outside the view. Then I'm stuck, potentially with UI elements at the bottom of the page being blocked by the address bar.

cc @hiikezoe

I feel like in this situation we should be sizing the app to fit into the space above the toolbar.

**Comment by Mugurell** [comment]
> I feel like in this situation we should be sizing the app to fit into the space above the address bar.

I was thinking.. would this then make the page a bit scrollable? Which could trigger a `INPUT_RESULT_HANDLED` based on which we'd try to animate the toolbar?

**Comment by Mugurell** [comment]
As an update for this, regarding edgecases in which we the toolbar might not be animated:
- pages use iframes/scrollable divs to display content 
- pages have their own scroll listeners
- pages have a height bigger than (window - toolbar) but smaller than window (ending underneath the toolbar), like I think is the case here - https://github.com/mozilla-mobile/fenix/issues/10477

@theres-waldo should we file individual GV tickets for these?

**Comment by theres-waldo** [comment]
> As an update for this, regarding edgecases in which the toolbar might not be animated:
>    * pages use iframes/scrollable divs to display content

This is only a problem if we're in the third case "pages have a height bigger than (window - toolbar) but smaller than window (ending underneath the toolbar)", and the solutions for that will apply to this case as well.

>    * pages have their own scroll listeners

First a small clarification: the listeners that are of interest to us are listeners for input events for which the default browser action is to scroll. These are generally `wheel`, `touch`, and `pointer` events. They are interesting because they can be `preventDefault()`ed by web content in which case the browser does not scroll. (There are also `scroll` events, which are more like a notification to web content that scrolling _has_ been performed. They cannot  be `preventDefault()`ed (the scrolling has already happened) and are not interesting for this discussion.)

In most cases, when a web app has a listener which `preventDefault()`s these events, the page does not actually have scrollable content (that is, the page's content size does not exceed the viewport size). For example, a mapping application will typically react to such an event by drawing new content onto a fixed-size map canvas. In these cases, there should be no problem related to the toolbar (as of [bug 1633322](https://bugzilla.mozilla.org/show_bug.cgi?id=1633322)).

It's possible that a page _has_ scrollable content (content size exceeds viewport size), _and_ `preventDefault()`s these events, _and_ performs scrolling manually (e.g. via `window.scrollTo()` in response to these events). I think such pages are (very) rare and I would consider them badly designed, but for these we would run into the "content obscured by toolbar" problem. It's worth filing a GV bug about this, though I don't have a great solution in mind for it :/ (But again, it may be rare enough that it doesn't matter.)

>   * pages have a height bigger than (window - toolbar) but smaller than window (ending underneath the toolbar), like I think is the case here - #10477

Yep, I believe this covers almost all of the problematic scenarios remaining now that [bug 1633322](https://bugzilla.mozilla.org/show_bug.cgi?id=1633322) is fixed, and it would be good to have a GV ticket on file for this.

My suggestion for how to handle this case, as mentioned, is to size the viewport to the (window - toolbar) height. The resulting behaviour would depend on the page:

 * If the page has a fixed content height (e.g. a GitHub issues list with a fixed number of issues to display) that just happens to fall into this small range, then the page would become a bit scrollable, like you said.
 * If the page has content height == viewport height, then the content height would shrink. This covers cases like the ublock settings page (where an iframe's viewport takes up the page viewport; the iframe would remain scrollable with the smaller viewport, as desired), as well as cases like a mapping or drawing application where the canvas size would simply be reduced by the height of the toolbar.

**Comment by mattok** [comment]
> I think most of the issues here should have been resolved in the latest Nightly by https://bugzilla.mozilla.org/show_bug.cgi?id=1633322
> ...
> @ QA Have you seen this still reproducing in latest nightly?

- Issue #14421 persists.
- With regards to thetimes.co.uk, I found that I could reproduce this on standard release firefox, but only when I do the following. Zoom, and then unzoom. It is not possible to fully unzoom, so the bottom of the page and associated buttons are inaccessible, and you are stuck. On Firefox Nightly, it is sometimes possible to unzoom, but zoom is extremely buggy almost to the point of being unusable: Very slow, parts of the screen go black, sometimes nothing seems to happen at all when you try to zoom, rotating the screen only makes matters worse etc.

**Comment by mattok** [comment]
>     * With regards to thetimes.co.uk, I found that I could reproduce this on standard release firefox, but only when I do the following. Zoom, and then unzoom. It is not possible to fully unzoom, so the bottom of the page and associated buttons are inaccessible, and you are stuck. On Firefox Nightly, it is sometimes possible to fully unzoom, but zoom is extremely buggy almost to the point of being unusable: Very slow, parts of the screen go black, sometimes nothing seems to happen at all when you try to zoom, rotating the screen only makes matters worse etc.

I wanted to send you a screenshot of the blacking out of parts of the screen on Firefox nightly on zoom/unzoom - but the screen appears to repaint and fix itself when I take a screenshot - so I can't do that! Anyway, perhaps the zooming problem is a separate issue.

**Comment by mattok** [comment]
Here is a [minimal example](https://codepen.io/mattok/full/XWdeQZY) to demonstrate issue #14421. If you can't see the yellow sticky bottom zone at the bottom because it's hidden under the address bar, there's a problem.

**Comment by mattok** [comment]
@Mugurell  Please correct me if I'm wrong, but I think that the sticky-bottom issue (#14421) and associated minimal test case, for which the pages could be longer than viewport and you would still expect the position sticky items to be placed relative to the bottom of the _visible_ viewport (i.e. relative to the top of the address bar when it is placed at the bottom), is distinct from the scroll problem, which sounds like it only affects pages that are larger than the visible window but less than (visible window + toolbar). I don't know enough about how things work under-the-hood to know whether a single change would fix both (e.g. it's just that the visible viewport isn't being defined properly when the address bar is at the bottom and fixing this would fix both problems.)

**Comment by Mugurell** [comment]
@mattok I'm not sure also but this can be triaged in GeckoView - whether there is a single fix or separate issues to be resolved.

**Comment by mattok** [comment]
OK. I've created #14691 for the zoom issue.

**Comment by Mugurell** [comment]
Based on https://github.com/mozilla-mobile/fenix/issues/14377#issuecomment-683061025 the offerup issue is probably the same as the one reported by mattok in https://github.com/mozilla-mobile/fenix/issues/13797#issuecomment-686329820 so it will probably be resolved by the same https://bugzilla.mozilla.org/show_bug.cgi?id=1663000.
Looks like this ^^ is the common issue for the reports here.

**Comment by eliserichards** [comment]
https://github.com/mozilla-mobile/fenix/issues/15040 also reports a similar issue

**Comment by ShintoPlasm** [comment]
> > ```
> > * With regards to thetimes.co.uk, I found that I could reproduce this on standard release firefox, but only when I do the following. Zoom, and then unzoom. It is not possible to fully unzoom, so the bottom of the page and associated buttons are inaccessible, and you are stuck. On Firefox Nightly, it is sometimes possible to fully unzoom, but zoom is extremely buggy almost to the point of being unusable: Very slow, parts of the screen go black, sometimes nothing seems to happen at all when you try to zoom, rotating the screen only makes matters worse etc.
> > ```
> 
> I wanted to send you a screenshot of the blacking out of parts of the screen on Firefox nightly on zoom/unzoom - but the screen appears to repaint and fix itself when I take a screenshot - so I can't do that! Anyway, perhaps the zooming problem is a separate issue.

On thetimes.co.uk this problem occurs when I try to log into my account:
1. Go to www.thetimes.co.uk
2. Tap the hamburger 'Menu' in the top-left corner
3. Scroll all the way to the bottom (with bottom bar enabled)

The link titled 'LOGIN' is partially obscured by the address bar: see attached screenshot.
![photo_2020-10-25_23-13-39](https://user-images.githubusercontent.com/35544416/97121643-cadb7a00-1717-11eb-96b0-4a770646d999.jpg)

**Comment by ShintoPlasm** [comment]
The issue on thetimes.co.uk and some other websites has been fixed for me on Stable 82.1.1 as of today.

**Comment by ShintoPlasm** [comment]
The issue on thetimes.co.uk and some other websites has been fixed for me on Stable 82.1.1 as of today.

**Comment by Mugurell** [comment]
So with this I think we have all the issues here fixed, thank you all!
If you're still encountering any of these please leave a comment, if not I'll close this in a few days.

**Comment by mattok** [comment]
Issue #14421 has not been addressed by the latest nightly. Its associated minimal example <https://codepen.io/mattok/full/XWdeQZY> shows that position: sticky is still not being calculated relative to the visible viewport when the address bar is on the bottom. (See my previous comment about this: <https://github.com/mozilla-mobile/fenix/issues/13797#issuecomment-686544599>) Thanks.

**Comment by hiikezoe** [comment]
@mattok #14421 is one of the cases of https://github.com/w3c/csswg-drafts/issues/5646, it's quite unclear whether overflow:hidden on the root element clips to 100% height or 100vh height.  (And to me clipping to 100vh is weird) Anyway, I don't recommend using overflow:hidden on the root element with 100vh height elements.

**Comment by mattok** [comment]
> @mattok #14421 is one of the cases of [w3c/csswg-drafts#5646](https://github.com/w3c/csswg-drafts/issues/5646), it's quite unclear whether overflow:hidden on the root element clips to 100% height or 100vh height. (And to me clipping to 100vh is weird) Anyway, I don't recommend using overflow:hidden on the root element with 100vh height elements.

Thank you @hiikezoe, but unless I have missed something, you are mistaken. The minimal example does not make use of `overflow: hidden` at all. Therefore, I don't believe this categorisation is correct. The *only* CSS required is:

```css
.sticky-bottom
{
  position: sticky;
  bottom: 0;  
}
```

The `position: sticky` applies to elements that should be offset relative to the nearest scrolling ancestor - which in this case, is the root and it is the visible viewport which should be being scrolled here. Therefore, the element should get stuck to the bottom of the visible window on scroll. This is what happens if `top: 0` is used and what happens on the desktop version of Firefox, which has consistent behaviour for `top: 0` and `bottom: 0`.

I think the fundamental issue here is that behaviour at the bottom of the screen (and its interaction with the address bar) should be completely symmetrically with behaviour at the bottom of the screen (and its interaction with the address bar). From a user's perspective, the viewport is what can be seen and should never be considered to be under the address bar. Why should `position:sticky; top: 0` position the sticky elements with respect to the top of the visible viewport when scrolling down the page, whereas `position:sticky; bottom: 0`positions the sticky elements *underneath* the address bar when scrolling down the page?

See the following additional variant example I've just made, which includes a sticky top and a sticky bottom element. Note that this example fails on the latest Firefox nightly whether or not the address bar is configured to be at the top or the bottom:

https://codepen.io/mattok/full/ExyEJRp

(Select 'Change View' and select 'Editor View' to be able to see the HTML and CSS.) On Firefox for Android, the bottom of the page is *always* cut off, whether or not the address bar is at the top or the bottom, so the sticky bottom element is *never* visible at the bottom of the screen. Whereas, on desktop Firefox, it is always visible on screen until the point at which you have scrolled past it completely, which I believe should be the correct behaviour. (Hopefully the fact that codepen places it within an iframe is not confusing matters... but nevertheless, behaviour is different across mobile and desktop - which is a hint that something is amiss.)

I hope that helps.

**Comment by hiikezoe** [comment]
> (Hopefully the fact that codepen places it within an iframe is not confusing matters... but nevertheless, behaviour is different across mobile and desktop - which is a hint that something is amiss.)

That iframe is exactly the problem I mentioned. In the iframe the sticky target is the iframe's viewport, not the root viewport.

**Comment by mattok** [comment]
Okay. So I stand  corrected on that. Thanks. My minimal example was a red herring, which was affected by the iframe codepen put it in. I put the example html code below in a file on my mobile device and loaded it locally using the file protocol and confirm that it behaves as I would expect a) on the current version of Firefox and b) with Firefox nightly:

```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <title>test sticky</title>
    <style>
.content { height: 100vh; }
.sticky { position: sticky; }
.top { top: 0; background-color: red; }
.bottom { bottom: 0; background-color: green; }
    </style>
  </head>
  <body>
    <div class="content">Content</div>
    <div class="sticky top">Sticky top (Red)</div>
    <div class="content">Content</div>
    <div class="sticky bottom">Sticky bottom (Green)</div>
    <div class="content">Content</div>
  </body>
</html>
```
It does.

For me at least, this then leaves me with two outstanding issues:

* The minimal example was an attempt to distil the core issue that was making the website shown in the screenshot at the top of #14421 be broken. While the minimal example does in fact work fine after all, that website is still broken even in FF Nightly, so I've clearly missed some complexity in the CSS that is in fact necessary to provoke the problem when doing the distilling. I'll investigate and get back to you on that, hopefully with a better minimal example.
* To me at least, I can't believe that the codepen situation is right as it stands. While I accept that I perhaps need to look more closely at the CSS codepen are using to understand why this is happening, the problem I have is that it leaves a discrepancy between behaviour on FF for desktop and mobile that I don't think developers should have to worry about. Why is the bottom of the page, and so the bottom of the iframe hidden on FF for Android regardless of scrolling (and regardless of address bar position, I think), whereas on desktop, scrolling will expose all content?

**Comment by mattok** [comment]
I've done some further analysis of the website affected by #14421. While I haven't been able to distil a minimal example yet, I observe the following:

* Chrome for desktop and Chrome for Android behave differently.
  - On desktop, a footer is always visible at the bottom of the page and 'position: sticky' items within a central scrollable pane are always visible at the bottom of that pane. This is true even if you put the website into responsive mode via the developer tools.
  - On Android, the bottom of the page (footer and bottom part of the middle pane) are initially cut off, but once the middle pane is fully scrolled, the scroll continues to expose the footer and correctly positioned sticky items.
* Firefox for desktop and Firefox for Android behave differently.
  - On desktop, a footer is always visible at the bottom of the page and 'position: sticky' items within a central scrollable pane are always visible at the bottom of that pane. This is true even if you put the website into responsive mode via the developer tools.
  - On Android, the bottom of the page is simply always cut off. It is not possible to expose it by scrolling. This occurs whether the address bar is at the top or the bottom of the page. I believe that this is a regression of some kind, and that FF was behaving *correctly* with no cut off content, and with the same behaviour across both desktop and mobile for this site in Sep 2019.

To me, both browsers are broken in the sense that both ought to behave consistently across the desktop and mobile versions, whereas they don't. However, the behaviour in FF in this situation is worse because it results in inaccessible content on Firefox for Android only. It is possible that the CSS on this site is somehow invalid/non-standard, and that both browsers are doing their best in the circumstances. Regardless Chrome is doing better from the perspective of a user that just needs to be able access the content.

If a FF for android developer wants to know the precise URL of the site in question, I can send it by a direct message of some kind. In the meantime, I'll continue to try to distil a minimal example.

**Comment by mattok** [comment]
I've managed to create a minimal example. It turns out that the position sticky issue was a distraction and not the root cause. The actual problem appears to be related to `height: 100vh; overflow: hidden;` in combination with flex box. The situation in this example across FF and Chrome, desktop and Android is exactly as described in my previous post:

```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <title>test flex</title>
    <style>
* { margin: 0; }
body { height: 100vh; overflow-y: hidden; flex-direction: column; display: flex; }
main { overflow-y: scroll; background-color: grey; }
    </style>
  </head>
  <body>
    <header>Header</header>
    <main>
      <ol>
        <li>First item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Item</li><li>Last item</li>
      </ol>
    </main>
    <footer>Footer</footer>
  </body>
</html>
```

The key thing is that on FF for Android the footer is cut off in a way that is inaccessible, regardless of scroll.

**Comment by hiikezoe** [comment]
The case is one of cases that https://bugzilla.mozilla.org/show_bug.cgi?id=1663000 will handle.

**Comment by juroth** [comment]
First of all: Thank you for building such a nice browser, I really enjoy firefox mobile <3

It seems like I am late to the party, but I encountered the same problem (100vh in combination with CSS grid and then the bottom toolbar will overlap the content). Here's a small example:
https://cdpn.io/juroth/debug/BaKJpXE/VJrxxOvywxZr
https://codepen.io/juroth/pen/BaKJpXE

Happens in 82.1.1 (Build #2015770923)

**Comment by hiikezoe** [comment]
@juroth codepen is not a good place to upload contents related to the dynamic toolbar.
Anyway the case will be also fixed by https://bugzilla.mozilla.org/show_bug.cgi?id=1663000 .

**Comment by Poopooracoocoo** [comment]
hey @hiikezoe. is this issue related to https://github.com/uBlockOrigin/uBlock-issues/issues/1304? if not, do you know which one is?

**Comment by hiikezoe** [comment]
@Poopooracoocoo Yes, that's what I supposed. Does the uBlockOrigin still persist on latest Fenix nightly? If it's still there, can you please open a new bug on bugzilla.mozilla.org and attach an html to reproduce it?
Thanks!

**Comment by Poopooracoocoo** [comment]
@hiikezoe Yeah the issue still persists in Nightly. I thought that the issue only applied to add-on pages. I don't think an html would help. I don't even know what it'd be. I've seen the same issue in uBO reported across Firefox repos like AC: https://github.com/mozilla-mobile/android-components/issues/6933. I don't know if its worth making a bug on BMO.

**Comment by hiikezoe** [comment]
@Poopooracoocoo thanks for the info. It looks like the uBlockOrigin case is "height: 100vh; overflow: hidden;" on the root element, which is https://bugzilla.mozilla.org/show_bug.cgi?id=1663634, not yet fixed unfortunately.

**Comment by Poopooracoocoo** [comment]
thanks hiro. just went and edited my other comments to include a link to that bug

**Comment by Raffarti** [comment]
I'd like to point out it affects an application of mine written in Qt/WASM. Even when the bar is on top, the bottom of the page is not visible and if the bar is swiped it goes away, but the bottom of the page is cut and shows the theme background color. It was working fine with the old toolbar.

**Comment by hiikezoe** [comment]
@Raffarti Issues happening with dynamic toolbar moving are caused by various reasons, also it depends on the toolbar position. So for the top toolbar case, you will see more problems rather than the bottom case because of https://github.com/mozilla-mobile/android-components/issues/7718.

Anyway, it's hard to tell what the underlying problem is in your case without seeing it.

**Comment by yookoala** [comment]
Could this be a form of #15337? It is currently tracking as [bug 1650644](https://bugzilla.mozilla.org/show_bug.cgi?id=1650644) and is reported as "Fixed" around 1 month ago.

**Comment by hiikezoe** [comment]
@yookoala what do you mean?  I believe the case of  #15337 has been already fixed.  You mean this issue has been already fixed? To be honest I haven't tried the original site so I am not sure.

**Comment by Raffarti** [comment]
> Anyway, it's hard to tell what the underlying problem is in your case without seeing it.

Here the page https://raffarti.gitlab.io/simpleadvancedmetronome/SimpleAdvancedMetronome.html

**Comment by Poopooracoocoo** [comment]
@Raffarti by old toolbar were you referring to Fennec's toolbar?

**Comment by Raffarti** [comment]
> @Raffarti by old toolbar were you referring to Fennec's toolbar?

The one that was in use before this, I believe it was the same currently used by firefox focus.

**Comment by hiikezoe** [comment]
> > Anyway, it's hard to tell what the underlying problem is in your case without seeing it.
> 
> Here the page https://raffarti.gitlab.io/simpleadvancedmetronome/SimpleAdvancedMetronome.html

At first glance, that's exactly same case in https://github.com/mozilla-mobile/fenix/issues/13797#issuecomment-728719819 that means it's related to [a spec issue](https://github.com/w3c/csswg-drafts/issues/5646).

**Comment by mind-bending-forks** [comment]
See also https://github.com/w3c/csswg-drafts/issues/5777 and https://mind-bending-forks.github.io/css-viewport-percentage-length-units/, which makes a case that this is more of a browser design issue related to details of the the implementation of hide/reveal feature for toolbars on mobile devices than it is a CSS specification issue.

**Comment by massic80** [comment]
One more fan of this issue: I love the awesome bar in the bottom, but this bug makes chats impossible to interact with (the message box is covered by address bar).
Do you think this will be ever fixed? This risks to make Fenix lose many of us devoted users :)

**Comment by gcp** [comment]
>Do you think this will be ever fixed? 

The page that this was originally reported on works correctly now, as scrolling further down (even though the page doesn't move) hides the address bar. You should file a specific bug about the problem you're still seeing.

**Comment by akostadinov** [comment]
@hiikezoe , I don't see a necessity for Firefox UI to depend on CSS to position itself. e.g. it can show UI elements outside page content when at the bottom of the page. Spec may affect how things currently work, but it shouldn't mean Firefox UI can improve the way it works going forward.

**Comment by massic80** [comment]
Think of 100%-high pages: a chat, for instance. (I got one not working, but it needs authentication). How can you scroll down, in order to see message input?
And, above all, how can you justify a different page behavior with the toolbar in the top and in the bottom?

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

🚩 Flag added 

Blocked by GV

**Comment by ffxsam** [comment]
I can confirm this is still an issue. In our web app, we have a left nav that has a "Trash" link at the bottom, and it's impossible for me to access it.

**Comment by massic80** [comment]
Anyone having a publicly visible not-working example?

**Comment by Raffarti** [comment]
> Anyone having a publicly visible not-working example?

https://github.com/mozilla-mobile/fenix/issues/13797#issuecomment-742396689

**Comment by massic80** [comment]
Sorry, @Raffarti, I noticed your page but I didn't get what was wrong until now, that I'm seeting the footer on a desktop PC.

**Comment by mobd** [comment]
I'm also not able to see the footer on twitter, though I'm not sure it's the same issue since it happens with the toolbar on top as well

**Comment by Mugurell** [comment]
> I'm also not able to see the footer on twitter, though I'm not sure it's the same issue since it happens with the toolbar on top as well

Can you please test with https://firefox-ci-tc.services.mozilla.com/tasks/index/gecko.v2.mozilla-central.latest.mobile/android-aarch64-debug to see if the problem reproduces here also?
(It's a stripped down version of the browser, helps eliminate some potential causes).
In all cases a video with the issue you're encountering would help a lot.

**Comment by mobd** [comment]
@Mugurell It doesn't happen there.
If I open it with geckoview example I can see the footer:
![output_1](https://user-images.githubusercontent.com/66154603/121562336-a370bd80-ca19-11eb-9d7d-d352b8d7917e.gif)

But with firefox I can't:
![output_0](https://user-images.githubusercontent.com/66154603/121562565-de72f100-ca19-11eb-930c-69058c95ee26.gif)
(You have to be logged in for the footer to show at all)
I'm using nightly 91.0a1 (Build #2015815371), I'm not sure when this started happening

**Comment by boreq** [comment]
Are there any updates regarding this issue? Currently the best way to fix this issue that  I can think about is to specifically target Firefox mobile with a fix. This is not acceptable and is effectively reverting the entire web to the days of lines upon and lines of CSS code fixing old IE versions.

**Comment by hiikezoe** [comment]
@boreq the eggplant case is slightly different.  In the case there's `html,body { overflow:hidden; }` style so that the 100vh height elements are clipped the ICB height. I did open [a spec issue]( https://github.com/w3c/csswg-drafts/issues/5646) for the case and the CSSWG decided to add [a note](https://drafts.csswg.org/css-overflow-3/#overflow-propagation);

> Note: overflow: hidden on the root element **might** not clip everything outside the Initial Containing Block if the ICB is smaller than the viewport, which can happen on mobile

(I made "might" words bold)

Though we have [an open bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1663634) for the case, I am not reluctant to change our behavior since clipping to overflow-ed area by `overflow: hidden` is really weird.

**Comment by ffxsam** [comment]
What's the current status on this?

**Comment by Taknok** [comment]
Speaking for myself, I tested 5 websites mentioned here. All are ok now. But I do not know if it is due to my screen size (xhdpi).
![Screenshot_20211228-105509](https://user-images.githubusercontent.com/12498020/147554219-b093c488-461d-4d5e-8e69-5a8aee92b553.jpg)

**Comment by Mugurell** [comment]
Thank you @Taknok!

@ffxsam There were many changes in the meantime with many problems fixed.
If you are still seeing issues - other than the ones related to https://github.com/mozilla-mobile/fenix/issues/13797#issuecomment-917724223 please open a new ticket(s) and we can look into addressing each issue separately.

**Comment by mattok** [comment]
I went back and re-reviewed my contributions to this thread. I've taken another look at the example I presented in https://github.com/mozilla-mobile/fenix/issues/13797#issuecomment-720049301 in Nov 2020, which can be seen (with minor variations) here: https://bugzilla.mozilla.org/attachment.cgi?id=9187912. Compared to the other browsers I checked (which didn't include iOS I'm afraid), Firefox for Android is still buggy for this case the following respects:

1. White space is displayed below the footer. This is already covered by [Firefox core bug 1673517](https://bugzilla.mozilla.org/show_bug.cgi?id=1673517).
2. On Firefox Nightly 97.0a1 but not on Firefox Daylight 95.2.0, the interaction between page refresh and scrolling is broken as follows: With the address bar at the bottom, visit the example above. (Similar thing happens with address bar at top.) The address bar will be displayed. In a single swipe up, scroll down, allowing inertia to take the scrolling to almost the bottom of the page. The address bar will remain and the footer will not be shown. Then swipe down to scroll back up to the top of the page. Instead of scrolling, the page will refresh. Attempt to swipe down very slowly instead so as to scroll rather than refresh. This wont work. The page will still attempt to refresh. The page is stuck in refresh mode, even after refreshing the page. The only way to get out of this buggy state and to scroll back to the top is to swipe up (scroll down) instead. This makes the address bar disappear and reveals the bottom of the content - the footer. The page can then be scrolled normally again. I believe this is covered by https://github.com/mozilla-mobile/fenix/issues/16577

In other respects, the above example at least behaves consistently across Firefox and Chrome now. I still find it frustrating that desktop and mobile behave differently with respect to visibility of the footer though, with the footer always visible on desktop irrespective of window size or scroll, but the user only made aware of the existence of the footer on mobile if they scroll down right to the very end of the content. Attempting to change that would require more effort than I have available to contribute though.

**Comment by ffxsam** [comment]
I'm still seeing issues with the bottom bar blocking important content. The bottom bar seems to overlay content rather than push it up, which I think is what it should be doing. But.. it's also possible something is wrong in my CSS. 🙂 

See the screenshots below. One shows a screenshot from my Pixel where the Trash nav item is blocked. The other screenshot is one from Firefox (desktop) showing the nav item.

<img src="https://user-images.githubusercontent.com/12532733/147781807-5ac35b42-3d4a-4ee4-ae58-839edabe5df5.png" width="33%" height="33%">

<img src="https://user-images.githubusercontent.com/12532733/147781835-55615088-615b-4a38-a97e-5adfa3d8be3c.png" width="33%" height="33%">

**Comment by Mugurell** [comment]
@ffxsam There might be some edgecases which are still problematic and as such need their own investigations.
Can you please provide a minimal test case to reproduce this or a publicly available website where this is happening?

**Comment by ffxsam** [comment]
@Mugurell Our new app is still not open to public access yet, but if you contact me privately, I can give you a signup code. [Visit our site](https://www.reelcrafter.com/) and then contact me via the support widget at the bottom right. Thanks!

**Comment by johnaps94** [comment]
@Mugurell Hello! I am facing a problem that i think is closely related to this. Please look here for full video explanation
https://i.stack.imgur.com/LAHor.gif
And here i have made a thread on stackoverflow
https://stackoverflow.com/questions/70893056/mozilla-android-addressbar-conflict-with-sticky-button-on-bottom/

I cant corrent the situation with css, so using a resize event listener is unfortunately not an option, can you confirm that this is a bug?

**Comment by Mugurell** [comment]
> I cant correct the situation with css, so using a resize event listener is unfortunately not an option, can you kindly confirm that this is a bug?
Can you please open a new ticket here https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Layout for more investigation?

**Comment by johnaps94** [comment]
@Mugurell Okay thank you, i just did, here it is https://bugzilla.mozilla.org/show_bug.cgi?id=1753019

**Comment by elibroftw** [comment]
Here are two of my websites https://elijahlopez.ca/social and https://lenerva.com/store that I just added bottom mobile navigation to. Am I doing something wrong with my CSS? The menu bar is at the bottom if the content requires more than the address bar of scrolling or no scrolling, but not if the content is slightly longer than the view height.

This is a custom CSS only mobile navigation

[CSS](https://github.com/elibroftw/website/blob/master/static/css/base.css#L456)
[HTML](https://github.com/elibroftw/website/blob/master/templates/base.html#L25)

**Comment by elibroftw** [comment]
[Demo of bug](https://youtu.be/U5EPzAIVm64)

**Comment by Mugurell** [comment]
@elibroftw Can you please also open a new ticket here https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Layout for more investigation?

**Comment by elibroftw** [comment]
Background fixed doesn't work properly in Firefox compared to edge.

![Screenshot_20220616-115257.jpg](https://user-images.githubusercontent.com/21298211/174113409-f07cacd9-2b59-4bbe-a47b-fdc2341bd6ad.jpg)

The bottom works until animations are in play.

**Comment by hiikezoe** [comment]
If you mean background-attachment:fixed, it's https://bugzilla.mozilla.org/show_bug.cgi?id=1704631 .

**Comment by elYanuki** [comment]
this is still a problem for me since my page is a no scroll app that just uses js to change the content on swipe, the searchbar still hides my content and it doesnt go away since my page doesnt overflow aka scroll.. maybe change the behaviour for sites that dont have scroll so that the content shrinks like when you disable searchbar hiding in generall

**Comment by theres-waldo** [comment]
> this is still a problem for me since my page is a no scroll app that just uses js to change the content on swipe, the searchbar still hides my content and it doesnt go away since my page doesnt overflow aka scroll.. maybe change the behaviour for sites that dont have scroll so that the content shrinks like when you disable searchbar hiding in generall

Could you share an affected URL (or attach an affected testcase) please?

(Probably best to do it in a new issue, but feel free to comment here with a link to the new issue.)

**Comment by flora-le** [comment]
I'm making a website, I have this problem on Firefox mainly. Vivaldi doesn't always hide the address bar sometimes but idk why, anyway.

I had two div, fullscreen, with or without scroll, one was okay, the other not. The case I'm talking about is on mobile phone of course.

I found the fix to my problem. I had put **min-height 100vh** to the one that had the bottom hidden by the address bar.  I removed it then it finally worked. Idk why it only happens in Firefox though. 

I wonder if it matters, but both the DIV are **position fixed**, and either needed top/bottom both to 0, or top to 0 and height 100%.

**Comment by theres-waldo** [comment]
@flora-le my advice to you is the same as in my previous comment:

> Could you share an affected URL (or attach an affected testcase) please?
>
> (Probably best to do it in a new issue, but feel free to comment here with a link to the new issue.)

**Comment by gabrielluong** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1813882

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).

**Comment by Poopooracoocoo** [comment]
Why was this moved to BMO? I thought only specific issues were being moved there. @gabrielluong

**Comment by cadeyrn** [comment]
@Poopooracoocoo All open issues are beinig moved to Bugzilla, as announced in #26855.
